powersimdata/network/europe_tub/model.py

```diff
@@ -0,0 +1,140 @@
+import os
+import shutil
+from zipfile import ZipFile
+
+from powersimdata.network.constants.region.europe import (
+    abv2country,
+    abv2timezone,
+    interconnect2abv,
+)
+from powersimdata.network.helpers import (
+    check_and_format_interconnect,
+    interconnect_to_name,
+)
+from powersimdata.network.model import ModelImmutables
+from powersimdata.utility.helpers import _check_import
+
+pypsa = _check_import("pypsa")
+zenodo_get = _check_import("zenodo_get")
+
+
+class TUB:
+    """PyPSA Europe network.
+
+    :param str/iterable interconnect: interconnect name(s).
+    :param int reduction: reduction parameter (number of nodes in network). If None,
+        the full network is loaded.
+    :param bool overwrite: the existing dataset is deleted a new dataset is downloaded
+        from zenodo.
+    """
+
+    def __init__(self, interconnect, reduction=None, overwrite=False):
+        """Constructor."""
+        self.grid_model = "europe_tub"
+        self.interconnect = check_and_format_interconnect(
+            interconnect, model=self.grid_model
+        )
+        self.data_loc = os.path.join(os.path.dirname(__file__), "data")
+        self.zenodo_record_id = "3601881"
+        self.reduction = reduction
+
+        if overwrite:
+            self.remove_data()
+
+        self.retrieve_data()
+
+    def remove_data(self):
+        """Remove data stored on disk"""
+        print("Removing PyPSA-Eur dataset")
+        shutil.rmtree(self.data_loc)
+
+    def retrieve_data(self):
+        """Fetch data"""
+        zenodo_get.zenodo_get([self.zenodo_record_id, "-o", f"{self.data_loc}"])
+        with ZipFile(os.path.join(self.data_loc, "networks.zip"), "r") as zip_network:
+            zip_network.extractall(self.data_loc)
+
+    def build(self):
+        """Build network"""
+        path = os.path.join(self.data_loc, "networks")
+        if self.reduction is None:
+            network = pypsa.Network(os.path.join(path, "elec_s.nc"))
+        elif os.path.exists(os.path.join(path, f"elec_s_{self.reduction}.nc")):
+            network = pypsa.Network(os.path.join(path, f"elec_s_{self.reduction}.nc"))
+        else:
+            raise ValueError(
+                "Invalid Resolution. Choose among: None | 1024 | 512 | 256 | 128 | 37"
+            )
+        id2zone = {i: l for i, l in enumerate(network.buses.index)}
+        zone2id = {l: i for i, l in id2zone.items()}
+
+        if "Europe" in self.interconnect:
```

Comment:
Here and elsewhere we do this check `if "Europe" in self.interconnect:` to see if we have the whole region. It seems from `check_and_format_interconnect` that if this is the case, `self.interconnect` will always be `["Europe"]`; should we check more clearly `if self.interconnect == ["Europe"]:`?

Reply:
Good catch

Reply:
Done